src/dependency-manager/spec/utils/slugs.ts

```diff
@@ -34,6 +34,10 @@ export class Slugs {
   public static ComponentSecretDescription = 'must contain alphanumeric character ([a-z0-9A-Z]), could contain dashes (-), underscores (_), and alphanumerics between.';
   public static ComponentSecretRegexBase = `[a-zA-Z0-9_-]+`;
   public static ComponentSecretValidator = new RegExp(`^${Slugs.ComponentSecretRegexBase}$`);
+
+  public static ReservedNameDescription = `must contain only alphanumeric characters ([a-z0-9A-Z]), could contain dashes (-), underscores (_), and alphanumerics between; max length ${Slugs.SLUG_CHAR_LIMIT}`;
```

Comment:
I don't think underscores are allowed for k8s resource names. Hadn't considered this case though...

I also don't want to allow bypassing the regex enforcement for service names from ArchitectSlugValidator.